src/legacy/core_plugins/kibana/public/visualize/np_ready/editor/editor.js

```diff
@@ -663,23 +662,21 @@ function VisualizeAppController(
   }
 
   const unlinkFromSavedSearch = () => {
-    const searchSourceParent = searchSource.getParent();
-    const searchSourceGrandparent = searchSourceParent.getParent();
+    if (!searchSourceParent) {
+      return;
+    }
 
-    delete savedVis.savedSearchId;
-    delete vis.savedSearchId;
-    searchSourceParent.setField(
-      'filter',
-      _.union(searchSource.getOwnField('filter'), searchSourceParent.getOwnField('filter'))
-    );
+    const searchSourceGrandparent = searchSourceParent.getParent();
+    const currentIndex = searchSourceParent.getField('index');
 
-    stateContainer.transitions.unlinkSavedSearch(
-      searchSourceParent.getField('query'),
-      searchSourceParent.getField('filter')
-    );
-    searchSource.setField('index', searchSourceParent.getField('index'));
+    searchSource.setField('index', currentIndex);
```

Comment:
I'm still not sure, whether we should inherit the index of the parent source while unlinking,
since both the `searchSource index` and `searchSourceParent index` are equal always. At least, I didn't find a case, when they could be different.

Reply:
if you create visualization based on saved search, then searchsource won't have index field, but the parent will.
calling searchSource.getField('index') will of course show the correct index (as that walks up the tree)